Src/AmrCore/AMReX_Interp_C.H

```diff
@@ -84,6 +84,154 @@ face_linear_face_interp_z (int fi, int fj, int fk, int n, Array4<T> const& fine,
     }
 }
 
+//
+// Fill fine values with tangential interpolation of coarse data.
+// Operate only on faces that overlap -- i.e., only fill the fine faces that
+// make up each coarse face, leave the in-between faces alone.
+//
+template<typename T>
+AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE void
+face_cons_linear_face_interp (int i, int j, int k, int n, Array4<T> const& fine,
+                              Array4<T const> const& crse, Array4<int const> const& mask,
+                              IntVect const& ratio,  Box const& per_grown_domain, int dim) noexcept
+{
+    int ci = amrex::coarsen(i, ratio[0]);
+
+#if (AMREX_SPACEDIM == 1)
+    amrex::ignore_unused(per_grown_domain);
+    int cj = 0;
+#else
+    int cj = amrex::coarsen(j, ratio[1]);
+#endif
+
+#if (AMREX_SPACEDIM == 3)
+    int ck = amrex::coarsen(k, ratio[2]);
+#else
+    int ck = 0;
+#endif
+
+    if (dim == 0 && ci*ratio[0] == i) {
+        // Check solve mask to ensure we don't overwrite valid fine data.
+        if (!mask || mask(ci, cj, ck, n)) {
+            fine(i, j, k, n) = crse(ci, cj, ck, n);
+#if (AMREX_SPACEDIM >= 2)
+            if (cj != per_grown_domain.smallEnd(1) && cj != per_grown_domain.bigEnd(1) && ratio[1] > 1) {
```

Comment:
Is the testing necessary? The CoarsenBox function grows the coarsened box by one cell. If the testing is necessary, it might be safer to test with `>` and `<` rather than `!=`.